kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -459,6 +459,12 @@ public void proverChecksX(Module specModule, Module mainDefModule) {
     }
 
     public void structuralChecks(scala.collection.Set<Module> modules, Module mainModule, Option<Module> kModule, Set<String> excludedModuleTags) {
+        if (kompileOptions.backend.equals(Backends.HASKELL))
+            errors.addAll(stream(modules).flatMap(m -> stream(m.localSentences()).flatMap(s -> {
+                if (s instanceof Rule && s.att().contains(Att.ANYWHERE()))
+                    return Stream.of(KEMException.compilerError(Att.ANYWHERE() + " is not supported by the " + Backends.HASKELL + " backend.", s));
+                return Stream.empty();
+            })).collect(Collectors.toSet()));
```

Comment:
Could this be pulled out into its own function?

Reply:
Fair point.